File: core/documents/universal_document_extraction_engine.py
from __future__ import annotations

from typing import Any, Dict, List, Optional

from core.documents.document_structure_engine import (
    build_document_structure,
)
from core.documents.document_hierarchy_engine import (
    build_document_hierarchy,
)
from core.documents.citation_extraction_engine import (
    extract_citations,
)
from core.documents.reference_extraction_engine import (
    extract_references,
)
from core.documents.document_table_engine import (
    extract_document_tables,
)
from core.knowledge.document_knowledge_graph_engine import (
    build_document_knowledge_graph,
)
from core.presentation.presentation_extraction_engine import (
    extract_presentation_structure,
)
from core.spreadsheets.spreadsheet_extraction_engine import (
    extract_spreadsheet_structure,
)
from core.ir.document_runtime_ir import (
    compile_document_runtime_ir,
)


def extract_document_runtime(
    text: str,
    slides: Optional[List[Dict[str, Any]]] = None,
    workbook: Optional[Dict[str, Any]] = None,
) -> Dict[str, Any]:
    structure = build_document_structure(
        text,
    )

    hierarchy = build_document_hierarchy(
        structure,
    )

    citations = extract_citations(
        text,
    )

    references = extract_references(
        structure,
    )

    tables = extract_document_tables(
        text,
    )

    knowledge_graph = build_document_knowledge_graph(
        structure,
    )

    slide_payload = extract_presentation_structure(
        slides or [],
    )

    worksheet_payload = extract_spreadsheet_structure(
        workbook or {},
    )

    document_ir = compile_document_runtime_ir(
        structure=structure,
        hierarchy=hierarchy,
        citations=citations,
        references=references,
        tables=tables,
        knowledge_graph=knowledge_graph,
        slides=slide_payload,
        worksheets=worksheet_payload,
    )

    return {
        "structure": structure,
        "hierarchy": hierarchy,
        "citations": citations,
        "references": references,
        "tables": tables,
        "slides": slide_payload,
        "worksheets": worksheet_payload,
        "knowledge_graph": knowledge_graph,
        "document_ir": document_ir,
        "bounded": True,
    }

File: src/documents/universalDocumentExtractionEngine.ts
/**
 * Converted from Python: core/documents/universal_document_extraction_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { buildDocumentStructure } from "./documentStructureEngine.js";
import { buildDocumentHierarchy } from "./documentHierarchyEngine.js";
import { extractCitations } from "./citationExtractionEngine.js";
import { extractReferences } from "./referenceExtractionEngine.js";
import { extractDocumentTables } from "./documentTableEngine.js";
import { buildDocumentKnowledgeGraph } from "../knowledge/documentKnowledgeGraphEngine.js";
import { extractPresentationStructure } from "../presentation/presentationExtractionEngine.js";
import { extractSpreadsheetStructure } from "../spreadsheets/spreadsheetExtractionEngine.js";
import { compileDocumentRuntimeIr } from "../ir/documentRuntimeIr.js";

export function extractDocumentRuntime(text: any, slides: any = null, workbook: any = null): any {
  var structure: any = buildDocumentStructure(text);
  var hierarchy: any = buildDocumentHierarchy(structure);
  var citations: any = extractCitations(text);
  var references: any = extractReferences(structure);
  var tables: any = extractDocumentTables(text);
  var knowledge_graph: any = buildDocumentKnowledgeGraph(structure);
  var slide_payload: any = extractPresentationStructure(py.or2(slides, () => ([])));
  var worksheet_payload: any = extractSpreadsheetStructure(py.or2(workbook, () => ({})));
  var document_ir: any = compileDocumentRuntimeIr(structure, hierarchy, citations, references, tables, knowledge_graph, slide_payload, worksheet_payload);
  return {"structure": structure, "hierarchy": hierarchy, "citations": citations, "references": references, "tables": tables, "slides": slide_payload, "worksheets": worksheet_payload, "knowledge_graph": knowledge_graph, "document_ir": document_ir, "bounded": true};
}
export { buildDocumentHierarchy, buildDocumentKnowledgeGraph, buildDocumentStructure, compileDocumentRuntimeIr, extractCitations, extractDocumentTables, extractPresentationStructure, extractReferences, extractSpreadsheetStructure };
